modules/rostests/apitests/win32nt/ntuser/NtUserSetTimer.c

```diff
@@ -1,18 +1,293 @@
+
+    for (int i = 0; i < TEST2_COUNT; i++)
+    {
+        UINT_PTR locIndex = SetTimer(hwnd, i, TEST2_INTERVAL, NULL);
+        if (locIndex == 0)
+        {
+            countErrors++;
+        }
+    }
+
+    ULONGLONG startTime = GetTickCount();
+
+    MSG msg = {};
+    while (GetMessage(&msg, NULL, 0, 0))
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+
+        if (GetTickCount() - startTime >= SLEEP_TIME)
+        {
+            PostQuitMessage(0);
+        }
+    }
+
+    for (int i = 0; i < TEST2_COUNT; i++)
+    {
+        if ((timerId2[i].counter < minMessages) || (timerId2[i].counter > maxMessages))
+        {
+            countErrors++;
+        }
+    }
+
+    for (int i = 0; i < TEST2_COUNT; i++)
+    {
+        if (KillTimer(hwnd, i) == 0)
+        {
+            countErrors++;
+        }
+    }
+    DestroyWindow(hwnd);
+    UnregisterClass("TimerWindowClass", GetModuleHandle(NULL));
+
+    if (countErrors == 0)
+        return TRUE;
+    return FALSE;
+}
+
+BOOL test3(void)
+{
+    int countErrors = 0;
+
+    for (long i = 0; i < TEST3_COUNT; i++)
+    {
+        UINT_PTR locIndex = SetTimer(NULL, 0, TEST3_INTERVAL, TimerProc);
+        if (locIndex == 0)
+        {
+            countErrors++;
+        }
+        if (KillTimer(NULL, locIndex) == 0)
+        {
+            countErrors++;
+        }
+    }
+
+    if (countErrors == 0)
+        return TRUE;
+    return FALSE;
+}
+
+BOOL test4(void)
+{
+    int countErrors = 0;
+
+    WNDCLASS wc = {};
+    wc.lpfnWndProc = WindowProc;
+    wc.hInstance = GetModuleHandle(NULL);
+    wc.lpszClassName = "TimerWindowClass";
+    RegisterClass(&wc);
+
+    HWND hwnd = CreateWindowEx(0, "TimerWindowClass", "Timer Window", 0,
+        CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT,
+        HWND_MESSAGE, NULL, GetModuleHandle(NULL), NULL);
+
+    if (hwnd == NULL)
+    {
+        return FALSE;
+    }
+
+    for (int i = 0; i < TEST4_COUNT; i++)
+    {
+        UINT_PTR result = SetTimer(hwnd, 1, TEST4_INTERVAL, NULL);
+        if (result == 0)
+            countErrors++;
+        if (KillTimer(hwnd, 1) == 0)
+            countErrors++;
+    }
+
+    DestroyWindow(hwnd);
+    UnregisterClass("TimerWindowClass", GetModuleHandle(NULL));
+
+    if (countErrors == 0)
+        return TRUE;
+    return FALSE;
+}
+
+BOOL test5(void)
+{
+    int countErrors = 0;
+
+    UINT_PTR locIndex1 = SetTimer(NULL, 0, TEST1_INTERVAL, TimerProc);
+    if (locIndex1 == 0)
+        countErrors++;
+    if (KillTimer(NULL, locIndex1) == 0)
+        countErrors++;
+    UINT_PTR locIndex2 = SetTimer(NULL, 0, TEST1_INTERVAL, TimerProc);
+    if (locIndex2 == 0)
+        countErrors++;
+    if (KillTimer(NULL, locIndex2) == 0)
+        countErrors++;
+    if(locIndex1 == locIndex2)
+        countErrors++;
+    if (countErrors == 0)
+        return TRUE;
+    return FALSE;
+}
+
 START_TEST(NtUserSetTimer)
 {
+    // TEST WITH MESSAGES WITHOUT WINDOW - test count sended messages
```

Comment:
Suggestion:
  // TEST WITH MESSAGES WITHOUT WINDOW - test count of sent messages